File: 김용성/Lv2/짝지어제거하기.cpp
#include <iostream>
#include<string>
#include <vector>

using namespace std;

int solution(string s)
{
    int answer = -1;
    vector<char> v;
    v.push_back(s[0]);
    for(int i = 1; i < s.size(); i++)
    {
        v.push_back(s[i]);
        size_t size = v.size();
        if(v[size - 1] == v[size - 2])
        {
//            cout << "before : " << s << endl;
            v.pop_back();
            v.pop_back();
//            cout << "after : "<< s << endl;
        }
//        cout <<  endl;
    }
    if(v.size() == 0) answer = 1;
    else answer = 0;
    return answer;
}

File: 조선빈/Lv.2/짝지어 제거하기.py
def solution(s):
    answer = 0
    s = list(s)
    stack = []
    for i in range(len(s)):
        if len(stack) == 0:
            stack.append(s[i])
        else:
            if s[i] == stack[-1]:
                stack.pop()
            else:
                stack.append(s[i])
    if len(stack) != 0:
        answer = 0
    else:
        answer = 1
    return answer
